src/cider/examples/CiderHashTableExample.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <exec/operator/join/HashTableFactory.h>
+#include <gflags/gflags.h>
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+#include "exec/operator/join/CiderLinearProbingHashTable.h"
+#include "util/Logger.h"
+
+// utils for generate datas
+std::random_device rd;
+std::mt19937 gen(rd());
+
+int random(int low, int high) {
+  std::uniform_int_distribution<> dist(low, high);
+  return dist(gen);
+}
+
+// hashfunction would be in a sepearete hashfunction lib in future, right now is hardcode
+// only for int key
+struct MurmurHash {
+  size_t operator()(int64_t rawHash) {
+    rawHash ^= unsigned(rawHash) >> 33;
+    rawHash *= 0xff51afd7ed558ccdL;
+    rawHash ^= unsigned(rawHash) >> 33;
+    rawHash *= 0xc4ceb9fe1a85ec53L;
+    rawHash ^= unsigned(rawHash) >> 33;
+    return rawHash;
+  }
+};
+
+// Different data types should have different equal functions like float and double may
+// compare only partial precision Different data types. Equal functions would be in a
+// sepearete lib. This is INT equal function.
+struct IntEqual {
```

Comment:
Do we need an interface for users to impl they own equal methods? Meanwhile, we provide default impl.

Reply:
Yes. equal function should be a configurable. But  all equal functions I saw are using struct, so there may be some struct example without a virtual class as interface.